File: stacks_pipeline_analyses/analyses_AK_LowGenotypeRate/batch4/samfiles_coverage_calc.py
### This file with take the output of multiple `samtools depth` files and combine into a text file ###

individs = open("Korean_subset_individs.txt", "r")
outfile = open("KOR_catalogloci_depths.txt", "w")
outfile.write("sample\tn_cat_loci\tdepths\n")


sample_list_kor = []
for line in individs:
	sample_list_kor.append(line.strip())
individs.close()

catalog_loci_dict_kor = {}
depth_dict_kor = {}
n_cat_all_kor = []
depth_all_kor = []

for sample in sample_list_kor:
	infile = open("../../../PCod-Korea-repo/stacks_b7_wgenome/" + sample + "_depth.txt", "r")
	sample_cat_loci = []
	sample_depths = []
	for line in infile:
		linelist = line.strip().split()
		if linelist[0] not in sample_cat_loci:
			sample_cat_loci.append(linelist[0])
			sample_depths.append(int(linelist[2]))
	infile.close()
	catalog_loci_dict_kor[sample] = sample_cat_loci
	depth_dict_kor[sample] = sample_depths
	outfile.write(sample + "\t" + str(len(sample_cat_loci)) + "\t" + ",".join(str(x) for x in sample_depths) + "\n")

outfile.close()

File: stacks_pipeline_analyses/analyses_AK_LowGenotypeRate/batch4/samfiles_coverage_calc.R
library(readr)
PO010715_02_depth <- read_delim("D:/Pacific cod/DataAnalysis/PCod-Korea-repo/stacks_b7_wgenome/PO010715_02_depth.txt",
"\t", escape_double = FALSE, col_names = FALSE,
trim_ws = TRUE)

View(PO010715_02_depth)

sample_cat_loci = c()
sample_depths = c()
for (i in seq(1,3534522,1)) {
  catlocus = PO010715_02_depth[i,1]
  if ((catlocus %in% sample_cat_loci) == FALSE) {
    sample_cat_loci = c(sample_cat_loci, catlocus)
    depth = PO010715_02_depth[i,3]
    sample_depths = c(sample_depths, depth)
  }
}

output1 = paste("PO010715_02", length(sample_cat_loci), paste(sample_depths, collapse = ", "), sep = "\t")

outfile = "KOR_catalogloci_depths.txt"
cat(output1, file = outfile, append = TRUE)